### PR TITLE
Fix JitMinOptsTrackGCrefs for legacy x86 back-end

### PR DESCRIPTION
**Pull request by BruceForstall**
Addresses #12415

### CODE REVIEW

**Comment by BruceForstall** [comment]
@pgavlin @rartemev PTAL
cc @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test
@dotnet-bot test Tizen armel Cross Debug Build

File: src/jit/jitconfigvalues.h [hunk 1]
```diff
@@ -218,14 +218,15 @@ CONFIG_INTEGER(JitEnableNoWayAssert, W("JitEnableNoWayAssert"), 0)
 CONFIG_INTEGER(JitEnableNoWayAssert, W("JitEnableNoWayAssert"), 1)
 #endif // !defined(DEBUG) && !defined(_DEBUG)
 
-#if !defined(JIT32_GCENCODER)
-#if defined(_TARGET_AMD64_) && defined(FEATURE_CORECLR)
+// It was originally intended that JitMinOptsTrackGCrefs only be enabled for amd6 on CoreCLR. A mistake was
+// made, and it was enabled for x86 as well. However, it doesn't currently work with x86 legacy back-end, so
+// disable it for that. Whether it should continue to be enabled for x86 non-legacy-backend should be investigated.
```
Comment (pgavlin) [inline src/jit/jitconfigvalues.h]: Is there an issue tracking this investigation?

**Comment by pgavlin** [comment]
Was there some trouble getting things refactored s.t. we can include `jitconfig.h` late enough to properly observe all of the relevant preprocessor symbols?

File: src/jit/jitconfigvalues.h (hunk 1 above)
Comment (BruceForstall) [inline src/jit/jitconfigvalues.h]: #12415

**Comment by BruceForstall** [comment]
> Was there some trouble getting things refactored s.t. we can include jitconfig.h late enough to properly observe all of the relevant preprocessor symbols?

I haven't tried yet, as this change is smaller and fixes the issue at hand (legacy backend failures), but my initial investigation indicates it is non-trivial.

File: src/jit/jitconfigvalues.h (hunk 1 above)
Comment (pgavlin) [inline src/jit/jitconfigvalues.h]: Okay. Can you mention that issue in this comment?

**Pull request merged**